nixos/cinnamon: make it possible to remove core packages

bobby285271 opened the pull request:
###### Description of changes

This is https://github.com/NixOS/nixpkgs/pull/175620 but for Cinnamon.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

bobby285271 committed:
nixos/cinnamon: switch to xdg.mime.enable

We enable this option explicitly to prevent users from accidentally
disabling it and getting a broken desktop.

bobby285271 committed:
nixos/cinnamon: switch to xdg.icons.enable

We enable this option explicitly to prevent users from accidentally
disabling it and getting a broken desktop.

bobby285271 committed:
nixos/cinnamon: fix some typos

mkg20001 commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -123,14 +123,19 @@ in
+        cinnamon-screensaver
+        nemo
+        networkmanagerapplet # also nm-applet not needed
+      ] ++ utils.removePackagesByName [
```
not sure if removing some of these packages will accidently break the session. session requirements simply meant what the session desktop file lists as a requirement.

bobby285271 replied:
TBH I don't use excludePackages myself so :woman_shrugging: , but anyway now the list only includes accessibility package and theme package, hope it will make more sense

bobby285271 committed:
nixos/cinnamon: make it possible to remove more non-essential packages

Packages for accessibility and theme can now be removed with excludePackages

bobby285271 committed:
nixos/cinnamon: don't override greeter theme if the package is excluded

This makes it possible to remove mint themes in an easy way, in this
case we will just use the default from slick-greeter module, i.e. the
Adwaita theme.

bobby285271 commented:
@ofborg test cinnamon

bobby285271 commented:
Okay I am introducing typos in my "fix some typos" commit message, it is now fixed :-/

With https://github.com/NixOS/nixpkgs/pull/200167/commits/8380ac45386c8dcd9ea372eb761ce2ef74f71cbc it should be actually possible to remove mint themes in a single line, without having to touch the greeter settings.
